syntax = "proto3";

package todo;

message TodoComponent {
    string todo_name = 1;
    bool is_done = 2;
}

message ServerResponseComponent {
    bool is_success = 1;
    string message = 2;
}

message TodoCreateRequest {
    string todo_name = 1;
    string timestamp = 3;
}

message TodoCreateResponse {
    ServerResponseComponent response = 1;
    string timestamp = 2;
}

message TodoShowRequest {
    string timestamp = 1;
}

message TodoShowResponse {
    repeated TodoComponent todos = 1;
    string timestamp = 3;
}

message TodoUpdateRequest {
    TodoComponent todo = 1;
    string timestamp = 2;
}

message TodoUpdateResponse {
    ServerResponseComponent response = 1;
    string timestamp = 2;
}


service TodoGateway {
    rpc TodoCreate (TodoCreateRequest) returns (TodoCreateResponse) {}
    rpc TodoShow (TodoShowRequest) returns (TodoShowResponse) {};
    rpc TodoUpdate (TodoUpdateRequest) returns (TodoUpdateResponse) {}
}

